### ISSUE TITLE
Methods with generic parameters cannot be loaded

### ISSUE THREAD

**Issue by freya022** [classgraph/classgraph#694]
I am trying to load a method from a class with `MethodInfo#loadClassAndGetMethod`, but it's failing saying it didn't find the method, exception:

```
Exception in thread "main" java.lang.IllegalArgumentException: Method not found: com.freya02.botcommands.oldtests.ClassGraphBug.test
	at io.github.classgraph.MethodInfo.loadClassAndGetMethod(MethodInfo.java:849)
	at com.freya02.botcommands.oldtests.ClassGraphBug.main(ClassGraphBug.java:30)
```

The loading is failing because the method has a generic parameter, by debugging the internals i've found the parameter types used to find the method were wrong, if you have a single parameter which type is a generic, `MethodInfo#loadParameterClasses` returns the class the method is declared in, however, it should have returned the erasure of the generic parameter.

Here are the verbose logs as well as a reproducible example.
[CG-logs.log](https://github.com/classgraph/classgraph/files/9004324/CG-logs.log)
[ClassGraphBug.zip](https://github.com/classgraph/classgraph/files/9004331/ClassGraphBug.zip)

**Comment by lukehutch** [comment]
Hi @freya022,

Thanks for the bug report and reproducer testcase.

I think I fixed the bug, but please check whether it works for you, and please also look at the source code of my change to see if it works as you would expect it to.

Thank you!

**Comment by freya022** [comment]
Hello, seems like the issue is fixed, I don't have any issue with the source changes. Thanks.

**Comment by lukehutch** [comment]
Thanks for taking a look, released in 4.8.148.
